components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchFacetImpl.java

```diff
@@ -72,6 +75,32 @@ public SearchFacetImpl(final SearchService searchService,
   {
     this.searchService = checkNotNull(searchService);
     this.componentMetadataProducers = checkNotNull(componentMetadataProducers);
+    this.rebuildIndexLock = new ReentrantReadWriteLock();
+  }
+
+  @Override
+  @Guarded(by = STARTED)
+  public void rebuildIndex() {
+    rebuildIndexLock.writeLock().lock();
+    log.info("Rebuilding index of repository {}", getRepository().getName());
+    try {
+      searchService.deleteIndex(getRepository());
+      searchService.createIndex(getRepository());
+      // TODO: is this going to scale? Currently it returns a List...
+      try (StorageTx tx = facet(StorageFacet.class).openTx()) {
+        for (Component component : tx.browseComponents(tx.getBucket())) {
+          try {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)